tests/gui/models/flash-state.spec.js

```diff
@@ -441,6 +470,52 @@ describe('Browser: flashState', function() {
 
     });
 
+    describe('.getFlashUuid()', function() {
+
+      const UUID_REGEX = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$/;
+
+      it('should be initially undefined', function() {
+        m.chai.expect(flashState.getFlashUuid()).to.be.undefined;
+      });
+
+      it('should be a valid uuid if the flashing flag is set', function() {
+        flashState.setFlashingFlag();
+        const uuid = flashState.getFlashUuid();
+        m.chai.expect(UUID_REGEX.test(uuid)).to.be.true;
+      });
+
+      it('should return different uuids every time the flashing flag is set', function() {
+        flashState.setFlashingFlag();
+        const uuid1 = flashState.getFlashUuid();
+        flashState.unsetFlashingFlag({
+          sourceChecksum: '1234',
+          cancelled: false
+        });
+
+        flashState.setFlashingFlag();
+        const uuid2 = flashState.getFlashUuid();
+        flashState.unsetFlashingFlag({
+          sourceChecksum: '1234',
+          cancelled: false
+        });
+
+        flashState.setFlashingFlag();
+        const uuid3 = flashState.getFlashUuid();
+        flashState.unsetFlashingFlag({
+          sourceChecksum: '1234',
+          cancelled: false
+        });
+
+        m.chai.expect(UUID_REGEX.test(uuid1)).to.be.true;
+        m.chai.expect(UUID_REGEX.test(uuid2)).to.be.true;
+        m.chai.expect(UUID_REGEX.test(uuid3)).to.be.true;
+
+        m.chai.expect(uuid1).to.not.equal(uuid2);
+        m.chai.expect(uuid2).to.not.equal(uuid3);
```

Comment:
I guess for completion, you could also do `m.chai.expect(uuid3).to.not.equal(uuid1);` ;)